### PR TITLE
Raise more detailed exception when encountering invalid feature values

### LINKED ISSUE #92: Handle NaN or infinite feature values
Currently the user sees a generic "An error occurred while processing your request. Please try again at a later time." when trying to build a model using features that have invalid values (i.e., values that scikit-learn doesn't accept). Should either fail more verbosely or, preferably, retrain without those features and warn the user that they were omitted.

### PR DESCRIPTION
**Pull request by bnaul**
Fixes #92. Combined with cesium-ml/cesium_web#109 this should also make debugging failures on the frontend easier.

### CODE REVIEW

File: cesium/build_model.py [hunk 1]
```diff
@@ -81,9 +80,19 @@ def build_model_from_featureset(featureset, model=None, model_type=None,
...
+        if params_to_optimize:
+            model = fit_model_optimize_hyperparams(feature_df, featureset['target'],
+                                                   model, params_to_optimize, cv)
+        else:
+            model.fit(feature_df, featureset['target'])
+    except ValueError as e:
+        nan_feats = np.isnan(feature_df).any()
+        inf_feats = np.isinf(feature_df).any()
+        message = "Invalid feature values detected:"
+        if nan_feats.any():
```
Comment (acrellin) [inline cesium/build_model.py]: Does the call to `any()` above return an array of values?

**Comment by bnaul** [comment]
Yeah, it's `pd.DataFrame.any` which by default is column-wise.

**Comment by acrellin** [comment]
Looks good @bnaul 
:+1:

File: cesium/tests/test_build_model.py [hunk 2]
```diff
@@ -83,3 +59,29 @@ def test_fit_optimize():
...
+    fset.x_inf.values[0, 0] = np.inf
+    fset.x_nan.values[0, 0] = np.nan
+    model = build_model.MODELS_TYPE_DICT['RandomForestClassifier']()
+    try:
+        model = build_model.build_model_from_featureset(fset, model)
+    except ValueError as e:
+        assert 'x_valid' not in str(e)
+        assert 'x_inf' in str(e)
+        assert 'x_nan' in str(e)
+
```
Comment (stefanv) [inline cesium/tests/test_build_model.py]: Else?

**Comment by bnaul** [comment]
Huh?

**Comment by acrellin** [comment]
I think he means make sure that exception actually gets raised - if not, that test will pass.

**Comment by bnaul** [comment]
Ah, smart...how about now?

**Comment by acrellin** [comment]
:+1:

**Comment by stefanv** [comment]
🚂

**Pull request merged**